apollo-router/src/spec/selection.rs

```diff
@@ -268,6 +288,41 @@ pub(crate) enum Condition {
     Variable(String),
 }
 
+/// Returns the `if` condition and the `label`
+fn parse_defer(
+    directives: &[hir::Directive],
+    defer_stats: &mut DeferStats,
+) -> (Condition, Option<String>) {
+    for directive in directives {
+        if directive.name() == DEFER_DIRECTIVE_NAME {
+            let condition = Condition::parse(directive).unwrap_or(Condition::Yes);
+            match &condition {
+                Condition::Yes => {
+                    defer_stats.has_defer = true;
+                    defer_stats.has_unconditional_defer = true;
```

Comment:
why not `defer_stats.has_unconditional_defer &= true;` ? Is that condition meaning that there's at least one uncoditional defer, or that all of them are unconditional?

Reply:
At least one. I picked `has_` to mean "there exists".

`has_unconditional_defer` is used to handle the case where the variables for conditional defer are all false. If there also isn’t any unconditional defer, then that case doesn’t have defer at all and doesn’t need subselections.